regression/pages/ecommerce/create_coupon_page.py

```diff
@@ -0,0 +1,196 @@
+        self.q(css='.modal-header>.close').click()
+        self.wait_for_element_invisibility(
+            '#catalogModal',
+            'wait for catalog dialog box to close'
+        )
+        return courses_list
+
+    def select_coupon_type(self, coupon_type):
+        """
+        Provide coupon code type in the create coupons form
+        Args:
+             coupon_type:
+        """
+        select_value_from_drop_down(self, "code_type", coupon_type)
+
+    def select_category(self, category):
+        """
+        Provide coupon category in the create coupons form
+        Args:
+             category:
+        """
+        select_value_from_drop_down(self, "category", category)
+
+    def set_dates(self, start_date, end_date):
+        """
+        Provide start date and end date for the coupon
+        Args
+            start_date:
+            end_date:
+        """
+        fill_css_selectors = [
+            'input[name="start_date"]',
+            'input[name="end_date"]'
+        ]
+        fill_values = [start_date, end_date]
+        fill_input_fields(self, fill_css_selectors, fill_values)
+
+    def fill_email_domains(self, email_domains):
+        """
+        Provide comma separated email domains for restricting access
+        Args:
+         email_domains:
+        """
+        self.q(css='#email-domains').fill(email_domains)
+
+    def select_voucher_type(self, voucher_type):
+        """
+        Select voucher usage type
+        Args:
+             voucher_type:
+        """
+        select_value_from_drop_down(self, "voucher_type", voucher_type)
+
+    def fill_quantity(self, quantity):
+        """
+        Provide no of codes
+        Args:
+             quantity:
+        """
+        self.q(css='.form-control[name="quantity"]').fill(str(quantity))
+
+    def max_uses(self, max_uses):
+        """
+        Provide maximum no of uses
+        Args:
+            max_uses:
+        """
+        self.q(css='.form-control[name="max_uses"]').fill(str(max_uses))
+
+    def fill_coupon_benefit_value(self, benefit_value):
+        """
+        For discount coupons, provide discount value
+        Args:
+            benefit_value:
+        """
+        self.q(css='.form-control[name="benefit_value"]').fill(benefit_value)
+
+    def select_benefit_type(self, benefit_type):
+        """
+        Select discount type
+        Args:
+            benefit_type:
+        """
+        if benefit_type == "Absolute":
+            click_checkbox(self, '#benefit-fixed')
+        else:
+            click_checkbox(self, '#benefit-percent')
+
+    def fill_coupon_client(self, client):
+        """
+        Provide client name
+        Args:
+             client:
+        """
+        self.q(css="input[name='client']").fill(client)
+
+    def select_coupon_invoice_type(self):
+        """
+        Provide invoice type
+        """
+        click_checkbox(self, '.invoice-type>div>#not-applicable')
```

Comment:
This looks like you're selecting a specific invoice type,right? You should rename the method.